src/transformStateWithClones.js

```diff
@@ -7,7 +7,30 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let copy = { ...state };
+  const statusArray = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(copy, action.extraData);
+        break;
+
+      case 'removeProperties':
+
+        for (const key of action.keysToRemove) {
```

Comment:
```suggestion
        for (const key of action.keysToRemove) {
```
"keysToRemove is an array. You shouldn't name a variable as key, better to name it as element or smth like that. But it is just a suggestion